fix OSX64 ABI mismatch with divcoeff

MartinNowak opened the pull request:
- also apply fix from #6180 (Issue 16536) to divcoeff.c
- add comment that declarations must be kept in sync

WalterBright commented:
I wanted to keep divcoeff.c independent of all the rest of the declarations, hence the (small) duplication of declaration code. It is worth it. I recommend adding a comment and folding in the other change to cdef.h you'd made.

MartinNowak committed:
fix OSX64 ABI mismatch with divcoeff

- also apply fix from #6180 (Issue 16536) to divcoeff.c
- add comment that declarations must be kept in sync

MartinNowak commented:
Done @WalterBright.
I've seen the intention but didn't saw enough value in it to warrant manually keeping code in sync.

WalterBright commented:
I guess we're going to disagree on the value of that. `divcoeff.c` has nothing to do with the rest of the compiler - it's a name mangling issue between C++ and D.

MartinNowak commented:
Well divcoeff is supposed to work on target integer types.
Anyhow I just did the small change, can we merge?

MartinNowak commented:
The PR now does what was requested in https://github.com/dlang/dmd/pull/6185#issuecomment-252454444, going to merge this now, so I can build the next beta.

MartinNowak commented:
Auto-merge toggled on

WalterBright commented:
Yes, please merge. Thanks!

schveiguy commented:
Thanks @MartinNowak, just confirmed this now works for me.
